calc/src/mechanics/gen12.ts

```diff
@@ -74,7 +74,13 @@ export function calculateRBYGSC(
     }
   }
 
-
+  //in gen 1, Super fang can hit ghosts
+  if (gen.num === 1 && (move.named('Super Fang'))) {
+      const lostHP = Math.floor(defender.curHP() / 2) || 1;
+      result.damage = lostHP;
+      return result;
+  }
```

Comment:
In Gen 2, you should check if the opposing Pokemon can be hit by Super Fang (probably by `getMoveEffectiveness`) and run the calculation if so.

And also check if the opposing Pokemon has Protect up just like it's done in `gen789.ts`.

Reply:
That was my bad, i put the calculation of Super fang before the verification of type effectiveness

Reply:
In gen 2 the check for Super fang pass after the check of type effectiveness.

Reply:
The same goes for the field.defenderSide.isProtected check, so as far i can tell, there is no issue with that.